test/functional/testlib.bash

```diff
@@ -1986,10 +2007,40 @@ update_bundle() {
 
 	# re-order items on the manifest so they are in the correct order based on version
 	sudo sort -t$'\t' -k3 -s -h -o "$bundle_manifest" "$bundle_manifest"
-
 	update_manifest -p "$bundle_manifest" contentsize "$contentsize"
 	update_manifest -p "$bundle_manifest" timestamp "$(date +"%s")"
 
+	if [ "$iterative" = true ]; then
+		# create the iterative "to manifest"
+		to_manifest="$bundle_manifest".I."$oldversion"
+		# get the manifest header as is
+		to_manifest_content=$(sed "/^[^\\t]*\\t[^\\t]*$/! d" "$bundle_manifest")$'\n'$'\n'
+		# get only those bundles that match the latest version
+		to_manifest_content+=$(sed "/^....\\t.*\\t$version\\t.*$/! d" "$bundle_manifest")$'\n'
+		write_to_protected_file "$to_manifest" "$to_manifest_content"
+		update_manifest "$to_manifest" filecount "$(grep -c -E "^(F|D)..." "$to_manifest")"
+		# TODO(castulo): update the contentsize in the to_manifest
+		# update the MoM, first remove any other iterative manifest for that bundle if one exists
+		if [ -n "$(awk "/$bundle.I.*/"'{ print $4 }' "$version_path"/Manifest.MoM)" ]; then
+			remove_from_manifest -p "$version_path"/Manifest.MoM "$(awk "/$bundle.I.*/"'{ print $4 }' "$version_path"/Manifest.MoM)"
+		fi
+		# now add the iterative manifest
+		add_to_manifest -p "$version_path"/Manifest.MoM "$to_manifest" "$bundle".I."$oldversion"
+
+		# create the iterative "from manifest" (if there are deltas)
+		if [ -n "$from_manifest_content" ]; then
+			from_manifest="$version_path"/Manifest."$bundle".D."$oldversion"
+			if [ ! -e "$from_manifest" ]; then
+				write_to_protected_file "$from_manifest" "$(sed "/^[^\\t]*\\t[^\\t]*$/! d" "$oldversion_path"/Manifest."$bundle")"$'\n'$'\n'
+			fi
+			write_to_protected_file -a "$from_manifest" "$from_manifest_content"
+			update_manifest -p "$from_manifest" filecount "$(grep -c -E "^(F|D)..." "$from_manifest")"
+			update_manifest -p "$from_manifest" timestamp "$(date +"%s")"
+			# TODO(castulo): update the contentsize in the from_manifest
+			add_to_pack "$bundle" "$from_manifest" "$oldversion"
```

Comment:
@jwakre don't you think it could be a good idea to keep it in $version_path also so you can debug your test easier if something goes wrong?

If you need to debug your test and the file has been deleted it will force you to unpack the tar first to be able to see the delta manifest, and since the test environment is going to be deleted anyway having an extra file there should not be a problem IMO.

Reply:
Ok, this is fine with me.